modules/termination-state-aws/src/main/java/org/opencastproject/terminationstate/aws/AutoScalingTerminationStateService.java

```diff
@@ -0,0 +1,413 @@
+    this.lifecycleHeartbeatPeriod = OsgiUtil.getOptCfgAsInt(config, CONFIG_LIFECYCLE_HEARTBEAT_PERIOD).getOrElse(DEFAULT_LIFECYCLE_HEARTBEAT_PERIOD);
+    this.accessKeyIdOpt = OsgiUtil.getOptCfg(config, CONFIG_AWS_ACCESS_KEY_ID);
+    this.accessKeySecretOpt = OsgiUtil.getOptCfg(config, CONFIG_AWS_SECRET_ACCESS_KEY);
+  }
+
+  @Override
+  public void setState(TerminationState state) {
+    if (enabled && autoScaling != null) {
+      super.setState(state);
+
+      if (getState() != TerminationState.NONE) {
+        // As this might also be called via Endpoint terminate polling if required
+        if (lifecyclePolling) {
+          stopPollingLifeCycleHook();
+        }
+
+        // stop accepting new jobs
+        try {
+          String host = getServiceRegistry().getRegistryHostname();
+          getServiceRegistry().setMaintenanceStatus(host, true);
+        } catch (ServiceRegistryException | NotFoundException e) {
+          logger.error("Cannot put this host into maintenance", e);
+        }
+        startPollingTerminationState();
+      }
+    }
+  }
+
+  protected void startPollingLifeCycleHook() {
+    try {
+      // create and set the job. To actually run it call schedule(..)
+      final JobDetail job = new JobDetail(SCHEDULE_GROUP, SCHEDULE_LIFECYCLE_POLLING_JOB, CheckLifeCycleState.class);
+      job.getJobDataMap().put(SCHEDULE_JOB_PARAM_PARENT, this);
+      final Trigger trigger = TriggerUtils.makeSecondlyTrigger(lifecyclePollingPeriod);
+      trigger.setGroup(SCHEDULE_GROUP);
+      trigger.setName(SCHEDULE_LIFECYCLE_POLLING_TRIGGER);
+      scheduler.scheduleJob(job, trigger);
+      scheduler.start();
+      logger.info("Started polling for Lifecycle state change");
+    } catch (org.quartz.SchedulerException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected void stopPollingLifeCycleHook() {
+    try {
+      scheduler.deleteJob(SCHEDULE_GROUP, SCHEDULE_LIFECYCLE_POLLING_JOB);
+    } catch (SchedulerException e) {
+      // ignore
+    }
+  }
+
+  public static class CheckLifeCycleState implements Job {
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+      AutoScalingTerminationStateService parent = (AutoScalingTerminationStateService) context.getJobDetail().getJobDataMap().get(SCHEDULE_JOB_PARAM_PARENT);
+      if (parent.autoScaling != null) {
+        DescribeAutoScalingInstancesRequest request = new DescribeAutoScalingInstancesRequest().withInstanceIds(parent.instanceId);
+        DescribeAutoScalingInstancesResult result = parent.autoScaling.describeAutoScalingInstances(request);
+        List<AutoScalingInstanceDetails> instances = result.getAutoScalingInstances();
+
+        if (!instances.isEmpty()) {
+          AutoScalingInstanceDetails autoScalingInstance = instances.get(0);
+
+          if ("Terminating:Wait".equalsIgnoreCase(autoScalingInstance.getLifecycleState())) {
```

Comment:
Constant should either be a class constant or (ideally) constructed from AWS provided constants.

Reply:
found an enum, not that is returned by the get()